lib/proxy/class-iputils.php

```diff
@@ -96,7 +96,7 @@ public static function check_ip4( $request_ip, $ip ) {
 	 * @throws \RuntimeException When IPV6 support is not enabled
 	 */
 	public static function check_ip6( $request_ip, $ip ) {
-		if ( ! ( ( extension_loaded( 'sockets' ) && defined( 'AF_INET6' ) ) || @inet_pton( '::1' ) ) ) {
+		if ( ! ( ( extension_loaded( 'sockets' ) && defined( 'AF_INET6' ) ) || inet_pton( '::1' ) ) ) {
```

Comment:
No need to silence warnings here, if AF_INET6 is enabled, we have support for IPv6, and `::1` is a valid IPv6 address.

Reply:
How about when someone is running a slightly different PHP configuration locally?

Reply:
`AF_INET6` is defined when PHP is build with IPv6 support:
* [constant definition](https://heap.space/xref/PHP-7.4/ext/sockets/sockets.c?r=f1bf058d#725)
* `HAVE_IPV6` is defined if PHP is built [without `--disable-ipv6`](https://heap.space/xref/PHP-7.4/configure.ac?r=3938bfb5#971) and `AF_INET6` constant [defined by the system](https://heap.space/xref/PHP-7.4/configure.ac?r=3938bfb5#580)

`inet_pton()` [uses](https://heap.space/xref/PHP-7.4/ext/standard/basic_functions.c?r=32257ac1#3968) `HAVE_IPV6` to see if it should support IPv6 addresses.

Reply:
But does mu-plugins make PHP with IPv6 support mandatory, even for a local environment?

Reply:
Oh, I see what you mean. Yes, you are right. However, PHP does not spit a warning for invalid addresses:

```bash
for ver in 7.3 7.4 8.0; do docker run -it --rm php:$ver php -r "error_reporting(-1); inet_pton('this is not an address');"; done
```